File: frontend/demo/component/combobox/combo-box-item-class-name.ts
import 'Frontend/demo/init'; // hidden-source-line
import '@vaadin/combo-box';
import { html, LitElement } from 'lit';
import { customElement, state } from 'lit/decorators.js';
import { applyTheme } from 'Frontend/demo/theme';

@customElement('combo-box-item-class-name')
export class Example extends LitElement {
  protected override createRenderRoot() {
    const root = super.createRenderRoot();
    applyTheme(root);
    return root;
  }

  @state()
  private items = ['Apple', 'Banana', 'Orange', 'Pear'];

  // tag::snippet[]
  protected override render() {
    return html`
      <vaadin-combo-box
        label="Fruit"
        .items="${this.items}"
        .itemClassNameGenerator="${this.classNameGenerator}"
      ></vaadin-combo-box>
    `;
  }

  protected classNameGenerator(item: string): string {
    switch (item) {
      case 'Apple':
        return 'coral';
      case 'Banana':
        return 'gold';
      case 'Orange':
        return 'orange';
      case 'Pear':
        return 'yellowgreen';
      default:
        return '';
    }
  }
  // end::snippet[]
}

File: src/main/java/com/vaadin/demo/component/combobox/ComboBoxItemClassName.java
package com.vaadin.demo.component.combobox;

import java.util.List;

import com.vaadin.flow.component.combobox.ComboBox;
import com.vaadin.flow.component.html.Div;
import com.vaadin.flow.router.Route;
import com.vaadin.demo.DemoExporter; // hidden-source-line

@Route("combo-box-item-class-name")
public class ComboBoxItemClassName extends Div {

    public ComboBoxItemClassName() {
        // tag::snippet[]
        ComboBox<String> comboBox = new ComboBox<>("Fruit");
        comboBox.setItems(List.of("Apple", "Banana", "Orange", "Pear"));
        comboBox.setClassNameGenerator((item) -> {
            switch (item) {
            case "Apple":
                return "coral";
            case "Banana":
                return "gold";
            case "Orange":
                return "orange";
            case "Pear":
                return "yellowgreen";
            default:
                return "";
            }
        });
        // end::snippet[]
        add(comboBox);
    }

    public static class Exporter extends DemoExporter<ComboBoxItemClassName> { // hidden-source-line
    } // hidden-source-line
}
